### ISSUE TITLE
new features

### ISSUE THREAD

**Issue by aster94** [aster94/logic-analyzer#7]
@sancho11

We need to decide which formattation of the code we will keep because it is becoming a mess of whitespaces, wrong indentation ecc. Which IDE are you using for processing and which one for arduino?
I am using vscode for arduino and the default formatting and for processing the default IDE but the default formatting is not very good, maybe is better to use all on vscode this way the formatting between arduino and processing will be the same

I also made a todo list
- [ ] formatting
- [x] use of `pinMode` and `digitalWrite` outside time-critical task. This will improve readability and won't change the efficency
- [x] use of the `LED_BUILTIN` for readability and easier port to new boards
- [ ] pass the number of samples directly from processing (this way we won't need to modify the MCU sketches, all the configuration will be in processing)
- [ ] in the code comments use english
- [ ] the steps of the mouse wheel when we are in a deep zoom is too big, we should change the steps according to the `reducer`
- [ ] new images and update readme

What do you think about?

**Comment by sancho11** [comment]
Hi!, well yeah is a mess xD. I was using Notepad++ because its a ligth text editor with some programing tools, but is not a IDE.
I think that would be wonderful make that improvements.
Just some things to keep in mind. The number of examples define the amount of memory that the board will use when you run the program, so I think that we can´t do that. We could make the asignation of memory in the board code and them say to the board how many samples we want to receive as long as they are less than those defined in the code, and this in turn would depend on the capabilities of the board.
I dont understand what is "comment out the debug print statemens" xD
For the rest I think I quite agree.
So... what IDE you want to use?
And... Where do you think I should start?
Have a nice day! (or night xD)

**Comment by aster94** [comment]
> I think that would be wonderful make that improvements.

muy bien 😄

> Hi!, well yeah is a mess xD. I was using Notepad++ because its a ligth text editor with some programing tools, but is not a IDE.
So... what IDE you want to use?

ok, if i am not wrong you are using an old computer so maybe VSCode would be too heavy, I will adjust the formattation and push the code, then you could try to follow the existing formattation so you won't need to install more stuff

> The number of examples define the amount of memory that the board will use when you run the program, so I think that we can´t do that. We could make the asignation of memory in the board code and them say to the board how many samples we want to receive as long as they are less than those defined in the code, and this in turn would depend on the capabilities of the board.

Exactly, you perfectly got what i was thinking about! 

> I dont understand what is "comment out the debug print statemens" xD
from this `print("something")` to `//print("something")` i already done it, i forgot to mark it 😅

**Comment by aster94** [comment]
ok with commit 82bbeac the formatting is all the same in all the sketches! you can download the [new branch](https://github.com/aster94/logic-analyzer/tree/new-features)

We can split the todo list and do one at the time to avoid merge conflicts. I can work on this: 

> pass the number of samples directly from processing (this way we won't need to modify the MCU sketches, all the configuration will be in processing)

could you please rewrite some comments in english and choose better names for the `cursora`, `cursoraf` and so on? I will do the same for the buttons names and for other variables (i choosed very ugly names ahahah)

**Comment by sancho11** [comment]
Ok, I will do that!

**Comment by sancho11** [comment]
Yes, github has me a little confused. I know it is an important tool, however I do not know how to use it yet. I did not know how to update my fork when you made the commits, and now that I did commit it does not let me update it. I think I will erase my fork and I will clone it directly from yours I will try to update correctly from now on.

**Comment by aster94** [comment]
ningùn problema! git is quite complex 😫

So could you please confirm me that this is what you did?

> you downloaded as a zip file the branch new-features from my repo and from it you started adding the changes then uploaded all in your repo and you made the pull request, correct?

I need to know this because otherwise i need to look for missing commits

**Comment by sancho11** [comment]
Yeah I do that! the noob way xD

**Comment by aster94** [comment]
normal man! i did it that way until the last month!

so after you deleted your remote repo and your working folder do something like this:
```
git clone https://github.com/aster94/logic-analyzer.git
git chekout pr
```

then to update your repo from the remote do (since you are starting from a new copy this is not needed now but later it will be)
`git pull`

and to update your remote from your local do
`git push`

then do a PR from github!

**Comment by sancho11** [comment]
Hi and thanks, I was able to synchronize the fork!

**Comment by aster94** [comment]
@sancho11 I started making small changes after lunch, i stopped only now, damned programming 😂
I reworked really a lot of stuff, the most noticeables changes are 
- splitted the code in three parts
- class `Button` and `Box`
- function `mouse_over_channel` and `mouse_over_button` you can see how much of repetitive code these two functions saved us!
- deleted some ugly names, added more comments

**There are some bugs, but the main part is working**

anyway do i remember wrong or once the reducer was changed with the mouse wheel instead of the mouse buttons?

still there is more work to be done especially to make the code more readable, i have difficulties to understand some functions 😂

I think that next time i will rework the pin assignment to make it smaller!

**Comment by sancho11** [comment]
"anyway do i remember wrong or once the reducer was changed with the mouse wheel instead of the mouse buttons?"
Yes, you got me! I changed that functionality because I think it's more common to need to move forward and back than zoom, so change them.
"do you have any idea why the firsts lines drawn are far from the border? this was something that happened before my changes and i still didn't investigate it"
Yes, that is my fault too... I add a signal offset in the board code, because I thought it would be convenient to leave this space, to delimit the starting point, because if the signal did not start long after time zero, now it always starts 125 micro seconds (or the value that has the offset defined in the board code), so that it looks a little more orderly. Or so I think.
"ahhh be careful this time to bring this changes to your repository, check that your pr branch is even with this! right now you should see `This branch is 2 commits behind aster94:pr'"
Yes, I will.
"The channel selection part is not working, i am not sure where the error is, the only part i think that i changed was this else because i thought it was in the wrong position, what do you think about it? https://github.com/aster94/logic-analyzer/blob/new-features/Computer_Interface/Computer_Interface.pde#L1035 is it correct? 🤔"
Yes, I will take care of that.

Im glad that you like the changes!

**Comment by aster94** [comment]
> Yes, you got me! I changed that functionality because I think it's more common to need to move forward and back than zoom, so change them.

> Yes, that was me too... I add a signal offset in the board code, because I thought it would be convenient to leave this space, to delimit the starting point, because if the signal did not start long after time zero, now it always starts 125 micro seconds (or the value that has the offset defined in the board code), so that it looks a little more orderly. Or so I think.

okkkkkk don't worry for now we can leave these changes and think more about them later
Once you fix the channel selector and improve the readability we will decide!

**Comment by sancho11** [comment]
The drawbacks were solved, for this I had to make minimal changes in some sections of the code.
I have also seen your contributions and I think it is ideal to use classes to sort the data and functions, I like what you get and I will try to apply this methodology in my functions too!
Have a nice day :D

**Comment by aster94** [comment]
#9 merged! great work sancho 😁
I have an exam soon so for a few of weeks I will be unable to make any changes but we are almost at the end!

**Comment by aster94** [comment]
Unfortunately for other duties coming I won't be able to complete the work on this 😥
maybe one day I will complete this, until then all pull request concerning this issue or other are welcomed!

**Comment by sancho11** [comment]
It's a shame. I hope you do well in what you are doing! I'm also busy but as soon as I can, I'll make more contributions to the project.

**Comment by aster94** [comment]
thanks sancho! unfortunately I have the italian version of the "MIR examen" so until the summer i would be completely focused on it!

**Comment by sancho11** [comment]
I'm not from Spain (I'm from Latin America), I had to google it, haha.
I wish you the best of luck, you almost succeed, give the best of you!
Regards!

**Comment by aster94** [comment]
thanks Sancho, I will write back 😃

**Comment by truglodite** [comment]
Best of luck aster94, hope you do well!

To all, I've got some time to work on moving the documentation from computer_interface.pde to the readme. I've forked the repo and am keeping up to date with the development branch. I'll submit any PR's from there.
